components/search_bar/search_bar.jsx

```diff
@@ -81,15 +83,19 @@ export default class SearchBar extends React.PureComponent {
         }
     }
 
-    componentWillReceiveProps(nextProps) {
-        const {searchTerms} = nextProps;
+    static getDerivedStateFromProps(props) {
+        const {searchTerms} = props;
 
-        this.setState({
-            visibleSearchHintOptions: this.determineVisibleSearchHintOptions(searchTerms),
-        });
+        if (searchTerms !== SearchBar.lastSearchTerms) {
+            SearchBar.lastSearchTerms = searchTerms;
```

Comment:
Thanks for making those changes.

Could you also put the `lastSearchTerms` into the state? That's consistent with how we do it elsewhere. Having it in the state instead of as a static field is a bit more idiomatic, and it prevents multiple instances of the component with interfering with each other.

Reply:
Here it is the new commit!